### PR TITLE
write_dirs: set_partition expects zero-based partition indeces

### PR DESCRIPTION
**Pull request by avg-I**
See #4446

### CODE REVIEW

File: tests/zfs-tests/tests/functional/write_dirs/setup.ksh [hunk 1]
```diff
@@ -51,4 +51,7 @@ DISK=${DISKS%% *}
 
 log_must set_partition $SLICE "" $SIZE $DISK
 
+if is_linux; then
+	export SLICE=1
+fi
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/write_dirs/setup.ksh]: That's subtle.  Does this now mean we can enable `write_dirs_002_pos` in the runfile?  If so you should update the runfile accordingly in this patch.

Comment (avg-I) [inline tests/zfs-tests/tests/functional/write_dirs/setup.ksh]: Originally this was only to make `write_dirs/setup` succeed with a real disk and `write_dirs_001_pos` to pass.
I've just tried `write_dirs_002_pos` and it caused the following:
```
[335174.862685] Kernel panic - not syncing: hung_task: blocked tasks
```
So, no change here.

Comment (behlendorf) [inline tests/zfs-tests/tests/functional/write_dirs/setup.ksh]: OK, got it.  Then this LGTM.  You got really unlucky and hit a bunch of unrelated issues in the automated but nothing to caused by this.  I'll merge it shortly.

**Comment by behlendorf** [comment]
Merged as:

72a780c write_dirs: set_partition expects zero-based partition indeces

**Pull request closed without merging**